File: src/main/java/burrow/furniture/hoard/command/EntryCommand.java
package burrow.furniture.hoard.command;

import burrow.core.command.Command;
import burrow.core.command.CommandContext;
import burrow.core.command.CommandType;
import burrow.furniture.hoard.HoardFurniture;
import org.jetbrains.annotations.NotNull;
import picocli.CommandLine;

@CommandLine.Command(
    name = "entry",
    description = "Find an entry by its associated ID and display it."
)
@CommandType(HoardFurniture.COMMAND_TYPE)
public class EntryCommand extends Command {
    @CommandLine.Parameters(index = "0")
    private Integer id;

    @CommandLine.Option(
        names = {"-o", "--object"},
        description = "Display it as an entry object.",
        defaultValue = "false"
    )
    private Boolean asObject;

    public EntryCommand(@NotNull final CommandContext commandContext) {
        super(commandContext);
    }

    @Override
    public Integer call() {
        final var hoardFurniture = use(HoardFurniture.class);
        final var entry = hoardFurniture.getHoard().get(id);

        if (asObject) {
            final var hoard = hoardFurniture.getHoard();
            final var entryObject = hoard.getEntryObject(entry);
            final var environment = CommandContext.Hook.environment.getNotNull(commandContext);
            final var string = hoard.format(id, entryObject, environment);
            buffer.append(string);
        } else {
            buffer.append(hoardFurniture.entryToString(entry, commandContext));
        }

        return CommandLine.ExitCode.OK;
    }
}

File: src/main/kotlin/furnishing/hoard/command/EntryCommand.kt
package burrow.furnishing.hoard.command

import burrow.furnishing.hoard.Entry
import burrow.furnishing.hoard.EntryNotFoundException
import burrow.furnishing.hoard.Hoard
import burrow.kernel.command.Command
import burrow.kernel.command.CommandData
import picocli.CommandLine

@CommandLine.Command(
    name = "entry",
    description = ["Finds an entry by its associated ID and displays it."]
)
class EntryCommand(data: CommandData) : Command(data) {
    @CommandLine.Parameters(index = "0")
    private var id: Int = 0

    @CommandLine.Option(
        names = ["-p", "--properties"],
        description = ["Displays it as an entry object."],
        defaultValue = "false"
    )
    private var asProperties: Boolean = false

    @Throws(EntryNotFoundException::class)
    override fun call(): Int {
        if (id <= 0) {
            stderr.println("Invalid entry ID: $id")
            return CommandLine.ExitCode.USAGE
        }

        val entry = use(Hoard::class)[id]
        if (asProperties) printProperties(entry)

        return CommandLine.ExitCode.OK
    }

    private fun printProperties(entry: Entry) {
        stdout.println("{")
        for ((key, value) in entry.properties) {
            stdout.println("    $key: \"$value\"")
        }
        stdout.println("}")
    }
}